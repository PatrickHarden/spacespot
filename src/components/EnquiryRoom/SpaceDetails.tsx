import React from 'react'
import { useSelector } from 'react-redux'

import PropertyDetailsLayout from 'components/PropertyDetails/PropertyDetailsLayout'
import spaceSelectors from 'services/space/selectors'

const SpaceDetails = () => {
  const space = useSelector(spaceSelectors.selectedSpace)
  const parent = useSelector(spaceSelectors.parentSpace)
  const regionAmenities = useSelector(spaceSelectors.regionAmenities)

  return (
    <PropertyDetailsLayout
      space={space}
      parent={parent}
      regionAmenities={regionAmenities}
      isEnquiry={false}
    />
  )
}

export default SpaceDetails
