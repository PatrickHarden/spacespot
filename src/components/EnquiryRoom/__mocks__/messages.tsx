export const messages = [
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message',
    sentAt: '2018-10-18T12:40:35.886+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message1',
    sentAt: '2018-10-18T12:40:43.547+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message2',
    sentAt: '2018-10-18T12:40:44.946+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message3',
    sentAt: '2018-10-18T12:40:45.678+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message4',
    sentAt: '2018-10-18T12:40:46.856+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message5',
    sentAt: '2018-10-23T14:17:59.662+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'jk',
    sentAt: '2018-10-28T10:13:59.805+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message6',
    sentAt: '2018-10-28T10:16:16.585+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '11:44',
    sentAt: '2018-10-28T10:45:02.592+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '12:34',
    sentAt: '2018-10-28T11:23:27.467+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'll',
    sentAt: '2018-10-28T11:29:37.057+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '12:30',
    sentAt: '2018-10-28T11:30:13.643+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'The message',
    sentAt: '2018-10-29T08:51:21.007+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'Another',
    sentAt: '2018-10-29T09:28:49.589+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'Other',
    sentAt: '2018-10-29T09:29:02.900+0000',
  },
]
