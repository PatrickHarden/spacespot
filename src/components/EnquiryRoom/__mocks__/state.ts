export const mockState: any = {
  enquiry: {
    data: {
      spaceId: 'SS_SPACE_665',
      peer1Id: '1',
      peer2Id: '2',
      chatId: 159,
      enquireStatus: 'ACCEPTED',
      negotiationId: 139,
      enquireId: 159,
      flex: false,
    },
  },
  negotiation: {
    data: {
      flexible: false,
      terms: [
        {
          termId: 5,
          value: 'MONTHLY',
        },
        {
          termId: 6,
          value: '2000',
        },
        {
          termId: 7,
          value: '2',
        },
        {
          termId: 8,
          value: '2000',
        },
        {
          termId: 9,
          value: 'GUARANTEE',
        },
        {
          termId: 1,
          value: '3500',
        },
        {
          termId: 2,
          value: '40000',
        },
        {
          termId: 3,
          value: '2020-04-22',
        },
        {
          termId: 4,
          value: '13',
        },
      ],
      negotiationId: 139,
      enquireId: 159,
      createdAt: '2020-02-14T06:30:08.354+0000',
      lastUpdated: '2020-03-05T16:35:02.513+0000',
      lastModifyBy: '0062fb00-e389-4bb8-9119-613e3c45938a',
      status: 'PROPOSED',
      customFitout: false,
      fitoutOption: 0,
      specialProvision: 'gfd',
      specialProvisionStatus: 'CREATED',
    },
    loading: false,
    termsFields: {
      '1': 'Area',
      '2': 'Rent',
      '3': 'Start',
      '4': 'Duration',
      '5': 'Frequency',
      '6': 'AdditionalCosts',
      '7': 'Deposit',
      '8': 'ServiceCharges',
      '9': 'DepositType',
    },
    signerInfo: {
      id: 1,
      name: 'Name1',
      emailId: 'name1@email.com',
      companyName: 'companyA',
      companyNumber: '12345678',
    },
    customFitout: [],
  },
  user: {
    auth: {
      idToken: {
        claims: {
          // eslint-disable-next-line @typescript-eslint/camelcase
          extension_Landlord: true,
        },
      },
    },
  },
}
