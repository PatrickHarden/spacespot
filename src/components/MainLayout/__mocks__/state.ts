export const mockState: any = {
  user: {
    auth: {
      idToken: {
        claims: {
          // eslint-disable-next-line @typescript-eslint/camelcase
          extension_Landlord: true,
        },
      },
    },
  },
}
