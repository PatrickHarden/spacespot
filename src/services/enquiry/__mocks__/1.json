{
  "chatId": 1,
  "enquireId": 1,
  "spaceId": 2,
  "peer1Id": "8fa47f17-c44e-48eb-8e88-9e8c8fe5cda3",
  "peer2Id": "8fa47f17-c44e-48eb-8e88-9e8c8fe5cda4"
}
