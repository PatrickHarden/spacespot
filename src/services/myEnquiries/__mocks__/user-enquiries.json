[
  {
    "enquireId": 167,
    "negotiationId": 147,
    "enquireStatus": "PENDING_TO_ACCEPT",
    "spaceId": "SS_SPACE_650",
    "terms": [
      { "termId": 1, "value": "12" },
      { "termId": 2, "value": "12.0" },
      {
        "termId": 3,
        "value": "Tue Feb 25 2020 17:59:47 GMT+0100 (hora estándar de Europa central)"
      },
      { "termId": 4, "value": "123" },
      { "termId": 5, "value": "" },
      { "termId": 6, "value": "" },
      { "termId": 7, "value": "" },
      { "termId": 8, "value": "123.0" },
      { "termId": 9, "value": "" }
    ],
    "messages": [
      {
        "userId": "de2563f1-2e15-4a6a-9e39-9600a8ad4ec1",
        "message": "",
        "sentAt": "2020-02-25T16:59:53.122+0000"
      }
    ]
  },
  {
    "enquireId": 168,
    "negotiationId": 148,
    "enquireStatus": "PENDING_TO_ACCEPT",
    "spaceId": "SS_SPACE_650",
    "terms": [
      { "termId": 1, "value": "12" },
      { "termId": 2, "value": "12.0" },
      {
        "termId": 3,
        "value": "Tue Feb 25 2020 18:00:28 GMT+0100 (hora estándar de Europa central)"
      },
      { "termId": 4, "value": "13" },
      { "termId": 5, "value": "" },
      { "termId": 6, "value": "" },
      { "termId": 7, "value": "" },
      { "termId": 8, "value": "123.0" },
      { "termId": 9, "value": "" }
    ],
    "messages": [
      {
        "userId": "de2563f1-2e15-4a6a-9e39-9600a8ad4ec1",
        "message": "",
        "sentAt": "2020-02-25T17:00:34.014+0000"
      }
    ]
  }
]
