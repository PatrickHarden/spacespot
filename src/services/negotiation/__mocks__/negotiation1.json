{
  "terms": [
    { "termId": 1, "value": "34" },
    { "termId": 2, "value": "207" },
    { "termId": 3, "value": "29 Months" },
    { "termId": 4, "value": "200 Sq Mtr" },
    { "termId": 5, "value": "$5600 per month" },
    { "termId": 6, "value": "20th September" }
  ],
  "negotiationId": 129,
  "enquireId": 149,
  "createdAt": "2020-02-03T13:03:07.661+0000",
  "lastUpdated": "2020-02-11T09:06:15.764+0000",
  "lastModifyBy": "c700793a-1963-4b2b-996f-1939f853cc10",
  "status": "CREATED"
}
