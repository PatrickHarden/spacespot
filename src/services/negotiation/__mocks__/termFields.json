{
  "1": "Area",
  "2": "Rent",
  "3": "Start",
  "4": "Duration",
  "5": "Frequency",
  "6": "AdditionalCosts",
  "7": "Deposit",
  "8": "ServiceCharges"
}
