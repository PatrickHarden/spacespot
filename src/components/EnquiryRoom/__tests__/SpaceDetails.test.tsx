import React from 'react'
import { Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMockStore } from 'redux-test-utils'
import { Provider } from 'react-redux'

import SpaceDetails from '../SpaceDetails'
import { createBrowserHistory } from 'history'

import { mockState } from '../__mocks__/state'

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
  createIntlCache: () => jest.fn(),
  createIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

const mockDispatch = jest.fn()

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}))

it('Renders without crash', () => {
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Router history={createBrowserHistory()}>
      <Provider store={store}>
        <SpaceDetails />
      </Provider>
    </Router>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})
