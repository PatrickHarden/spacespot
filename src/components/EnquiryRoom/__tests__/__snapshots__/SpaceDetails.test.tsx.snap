// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crash 1`] = `
<DocumentFragment>
  <div
    class="sc-cMhqgX fxvhcV"
  >
    <div
      class="sc-bdVaJa coVgKN"
    >
      <div
        aria-live="polite"
        class="image-gallery"
      >
        <div
          class="image-gallery-content right"
        >
          <div
            class="image-gallery-slide-wrapper right"
          >
            <div
              class="image-gallery-slides"
            >
              <div
                class="image-gallery-slide  center "
                role="button"
                style="transform: translate3d(0%, 0, 0);"
                tabindex="-1"
              >
                <div>
                  <img
                    class="image-gallery-image"
                    src="building.svg"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="image-gallery-thumbnails-wrapper right"
          >
            <div
              class="image-gallery-thumbnails"
            >
              <div
                aria-label="Thumbnail Navigation"
                class="image-gallery-thumbnails-container"
                style="transform: translate3d(0, 0px, 0);"
              >
                <button
                  aria-label="Go to Slide 1"
                  aria-pressed="true"
                  class="image-gallery-thumbnail active"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="image-gallery-thumbnail-inner"
                  >
                    <img
                      class="image-gallery-thumbnail-image"
                      src="building.svg"
                    />
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-bwzfXH bPITde"
    >
      <div
        class="sc-EHOje eURbOX"
      >
        <div
          class="sc-bZQynM flKQuj"
        />
        <div
          class="sc-ifAKCX dUjEzj"
        >
          <a
            href="#summary"
          >
            DETAILS_SUMMARY
          </a>
          路
          <a
            href="#details"
          >
            DETAILS_DETAILS
          </a>
          路
          <a
            href="#ammenities"
          >
            DETAILS_AMMENITIES
          </a>
          路
          <a
            href="#photos"
          >
            DETAILS_PHOTOS
          </a>
          路
          <a
            href="#location"
          >
            DETAILS_LOCATION
          </a>
        </div>
        <div
          class="sc-gzVnrw eAgMhc"
        >
          <div
            class="sc-bxivhb lgYJdu"
            style="visibility: hidden;"
          >
            <img
              alt="size"
              src="favorite.svg"
            />
            <span>
              DETAILS_NAV_ADD_FAVORITE
            </span>
          </div>
        </div>
        <div
          class="sc-gzVnrw eAgMhc"
        >
          <div
            class="sc-bxivhb lgYJdu"
            style="visibility: hidden;"
          >
            <img
              alt="calendar"
              src="share.svg"
            />
            <span>
              DETAILS_NAV_SHARE
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-iuJeZd kxxkKF"
    >
      <div
        class="sc-esOvli hqUlqr"
      >
        <div
          class="sc-lhVmIH iUQYqa"
        >
          <h1
            class="sc-btzYZH sc-bYSBpT knZOHX"
            id="summary"
          />
          <div
            class="sc-jtRfpW fxoqEV"
          />
          <div
            class="sc-dqBHgY isQODI"
          >
            <a
              href="#location"
            >
              <img
                alt="spot"
                src="spot.svg"
              />
            </a>
            <a
              href="#location"
            >
              <h5>
                DETAILS_SEE_MAP
              </h5>
            </a>
          </div>
          <div
            class="sc-kTUwUJ fLoQqh"
          >
            <div />
          </div>
          <h5
            class="sc-gxMtzJ iaBTeL"
          >
            DETAILS_LICENSED_FOR
          </h5>
          <div
            class="sc-dfVpRl fkiYrW"
          >
            DETAILS_OFFICE
          </div>
          <div
            class="sc-jbKcbu dgrgHP"
          >
            <h2
              class="sc-csuQGl sc-tilXH bfdqJg"
              id="details"
            >
              DETAILS_DESKS
            </h2>
            <h2
              class="sc-csuQGl sc-tilXH bfdqJg"
              id="servicedOffices"
            >
              DETAILS_SERVICED_OFFICES
            </h2>
          </div>
          <h2
            class="sc-csuQGl sc-elJkPf kFrdkZ"
            id="ammenities"
          >
            DETAILS_AMMENITIES
          </h2>
          <div
            class="sc-jwKygS iwQXZP"
            data-auto="details-ammenities"
          >
            <ul />
          </div>
          <h2
            class="sc-csuQGl sc-elJkPf kFrdkZ"
            id="photos"
          >
            DETAILS_PHOTOS
          </h2>
          <div
            class="sc-feJyhm uuvPf"
            data-auto="detail-photos"
          />
          <h2
            class="sc-csuQGl sc-elJkPf kFrdkZ"
            id="location"
          >
            DETAILS_LOCATION
          </h2>
          <div
            class="sc-iELTvK fQcFYt"
          >
            <h2 />
            <p />
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-iQNlJl cklsVe"
      data-auto="footer"
    >
      <div
        class="sc-bsbRJL hABnEO"
      >
        <div
          class="sc-gwVKww fYXqY"
        >
          <div
            class="sc-eTuwsz gDnOCV"
          >
            <a
              href="/"
            >
              <img
                alt="SpaceSpot Logo"
                height="28px"
                src="logo.svg"
              />
            </a>
          </div>
          <ul>
            <li>
              <a
                href="mailto:oslo@spacespot.com"
              >
                FOOTER_CONTACT
              </a>
            </li>
            <li>
              <a
                href="/en-no/privacy"
              >
                FOOTER_PRIVACY
              </a>
            </li>
            <li>
              <a
                href="/en-no/terms"
              >
                FOOTER_TERMS_OF_SERVICE
              </a>
            </li>
            <li>
              <a
                href="/en-no/cookie-policy"
              >
                FOOTER_COOKIE_POLICY
              </a>
            </li>
            <li>
              <a
                href="/en-no/about-us"
              >
                FOOTER_ABOUT_US
              </a>
            </li>
            <li>
              <a
                href="/en-no/list"
              >
                FOOTER_SEARCH
              </a>
            </li>
          </ul>
        </div>
        <div
          class="sc-hXRMBi jlWtlW"
        >
          <div
            class="sc-RefOD kaacNG"
          >
            <div
              class="sc-ibxdXY dbyYuq sc-iQKALj jTMrqw"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline sc-eXEjpC vXBRT"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby=" lang-dd"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                  id="lang-dd"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="sc-bwCtUz dQQDZX"
                  >
                    <img
                      alt="uk"
                      src="uk.svg"
                    />
                    <div
                      class="sc-hrWEMg gJWJWc"
                    >
                      English (UK)
                    </div>
                  </div>
                </div>
                <input
                  type="hidden"
                  value="en-UK"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <ul>
            <li>
              <a
                href="https://www.facebook.com/myspacespot"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="FOOTER_FACEBOOK"
                  height="30px"
                  src="facebook.svg"
                  width="17px"
                />
              </a>
            </li>
            <li>
              <a
                href="https://www.instagram.com/myspacespot"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="FOOTER_INSTAGRAM"
                  height="30px"
                  src="instagram.svg"
                  width="30px"
                />
              </a>
            </li>
            <li>
              <a
                href="https://www.linkedin.com/company/Spacespot"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="FOOTER_LINKEDIN"
                  height="30px"
                  src="linkedin.svg"
                  width="30px"
                />
              </a>
            </li>
          </ul>
          <div
            class="sc-epnACN euNvsi"
          >
            FOOTER_RIGHTS_RESERVED
          </div>
        </div>
      </div>
      <div
        class="sc-epnACN sc-hZSUBg fBNVLq"
      >
        FOOTER_RIGHTS_RESERVED
      </div>
    </div>
  </div>
</DocumentFragment>
`;
